### ISSUE TITLE
Doxywizard truncates standard input passed to doxygen process.

### ISSUE THREAD

**Issue by nitz** [doxygen/doxygen#9127]
**Describe the bug**
When using `DOT_IMAGE_FORMAT = svg`, `DOT_TRANSPARENT = YES` does not have any effect. `dot` outputs a svg with a solid object with a non-transparent background.

**Expected behavior**
The generated svg background object should either be transparent or not exist.

**Screenshots**
![image](https://user-images.githubusercontent.com/602691/154492422-e7a38361-8c9e-4af6-9761-59fddae5db99.png)

**To Reproduce**
[dot-transparent-ineffective-svg.zip](https://github.com/doxygen/doxygen/files/8089179/dot-transparent-ineffective-svg.zip)
Minimal example attached. To see affected output, run `doxygen`, then open `docs/html/group__a.html`. The generated docs use a custom `css` that colors the background of `.contents` divs purple, so you can see the white background on the generated SVG, as shown in the screenshot above.

**Version**
`1.9.3 (c0b9eafbfb53286ce31e75e2b6c976ee4d345473)`, on x64 Windows 10 Pro 21H2 (19044.1526)

**Stack trace**
N/A

**Additional context**
Due to the way SVGs are included in the HTML output as an `iframe`, styling with CSS causes the non-transparent background to be ineffective. The `dot` file generated doesn't set a `bgcolor` attribute on the entire graph, though I do see it trying to in the code. As an example, here: https://github.com/doxygen/doxygen/blob/e18f715eb55121a4219d00bc4d824cebf1fb504b/src/dotgroupcollaboration.cpp#L317-L340

The SVG XML produced from the dot without the bgcolor attribute with the creates a polygon with a `white` fill: 

```xml
<!--snipped-->
<polygon fill="white" stroke="transparent" points="-4,4 -4,-23 146,-23 146,4 -4,4"/>
<!--snipped-->
```

Manually removing that `<polygon />` produces the desired outcome:
![image](https://user-images.githubusercontent.com/602691/154494267-c9caf438-df08-4cf4-a9dd-8e6f62386436.png)

If I manually edit a dot file, to add the attribute:

```diff
--- group__a.dot        Thu Feb 17 08:59:12 2022
+++ group__a-modified.dot       Thu Feb 17 08:59:13 2022
@@ -1,5 +1,6 @@
 digraph "Group A"
 {
+  bgcolor="transparent"
   edge [fontname="Helvetica",fontsize="10",labelfontname="Helvetica",labelfontsize="10"];
   node [fontname="Helvetica",fontsize="10",shape=box];
   rankdir=LR;
````

The `<polygon />` isn't created in the output.

The PR #8826 looks like it might be a potential fix for this, as the `DOT_COMMON_ATTR` it adds looks like it would be a place to specify `bgcolor="transparent"`, but it is my feeling that the `DOT_TRANSPARENT` still should create that attribute itself, but for some reason isn't.

Cheers!

**Comment by albert-github** [comment]
Which version of `dot` do you use?

I tried with `dot` versions 2.38.0 and 2.40.1 and in both cases I see:

![image](https://user-images.githubusercontent.com/5223533/154506414-b3bed09c-d3d0-478f-a0ab-f7d0ee7aee76.png)

Regarding the purple backgroud of "Group B" this is another, different but  fundamental, question. Probably this should be explicitly set to a color (white).

**Comment by nitz** [comment]
I'm using the `dot` that came with the latest stable win-x64 binaries from Graphviz:

```
C:\Users\cmd>dot -v
dot - graphviz version 2.50.0 (20211204.2007)
libdir = "C:\Program Files\Graphviz\bin"
Activated plugin library: gvplugin_dot_layout.dll
Using layout: dot:dot_layout
Activated plugin library: gvplugin_core.dll
Using render: dot:core
Using device: dot:dot:core
The plugin configuration file:
        C:\Program Files\Graphviz\bin\config6
                was successfully loaded.
    render      :  cairo dot dot_json fig gdiplus json json0 map mp pic pov ps svg tk vml xdot xdot_json
    layout      :  circo dot fdp neato nop nop1 nop2 osage patchwork sfdp twopi
    textlayout  :  textlayout
    device      :  bmp canon cmap cmapx cmapx_np dot dot_json emf emfplus eps fig gif gv imap imap_np ismap jpe jpeg jpg json json0 metafile mp pdf pic plain plain-ext png pov ps ps2 svg tif tiff tk vml xdot xdot1.2 xdot1.4 xdot_json
    loadimage   :  (lib) bmp eps gif jpe jpeg jpg png ps svg
```

Took a peek through Graphviz's [changelog](https://gitlab.com/graphviz/graphviz/-/blob/main/CHANGELOG.md), but at a quick glance I don't see anything that looks like like a change that describes what we're seeing. I'm betting something must have changed somewhere along the line to produce the different behavior.

I'll see what an earlier version of `dot`/Graphviz does for me.

**Comment by nitz** [comment]
The earliest binaries directly from Graphviz's download page were for 2.44.1:

```
C:\Users\cmd>dot -v
dot - graphviz version 2.44.1 (20200629.0800)
libdir = "C:\Program Files\Graphviz\bin"
Activated plugin library: gvplugin_dot_layout.dll
Using layout: dot:dot_layout
Activated plugin library: gvplugin_core.dll
Using render: dot:core
Using device: dot:dot:core
The plugin configuration file:
        C:\Program Files\Graphviz\bin\config6
                was successfully loaded.
    render      :  cairo dot dot_json fig gd gdiplus json json0 map mp pic pov ps svg tk vml vrml xdot xdot_json
    layout      :  circo dot fdp neato nop nop1 nop2 osage patchwork sfdp twopi
    textlayout  :  textlayout
    device      :  bmp canon cmap cmapx cmapx_np dot dot_json emf emfplus eps fig gd gd2 gif gv imap imap_np ismap jpe jpeg jpg json json0 metafile mp pdf pic plain plain-ext png pov ps ps2 svg svgz tif tiff tk vml vmlz vrml wbmp xdot xdot1.2 xdot1.4 xdot_json
    loadimage   :  (lib) bmp eps gd gd2 gif jpe jpeg jpg png ps svg
```
And the output is the same:
![image](https://user-images.githubusercontent.com/602691/154510513-cbbfdcbc-5448-4390-96db-d97955e42354.png)

I found a copy of 2.38.0 [in their archive](https://www2.graphviz.org/Archive/stable/windows/):

```
C:\Users\cmd>dot -v
dot - graphviz version 2.38.0 (20140413.2041)
libdir = "C:\Program Files\Graphviz\bin"
Activated plugin library: gvplugin_dot_layout.dll
Using layout: dot:dot_layout
Activated plugin library: gvplugin_core.dll
Using render: dot:core
Using device: dot:dot:core
The plugin configuration file:
        C:\Program Files\Graphviz\bin\config6
                was successfully loaded.
    render      :  cairo dot fig gd gdiplus map pic pov ps svg tk vml vrml xdot
    layout      :  circo dot fdp neato nop nop1 nop2 osage patchwork sfdp twopi
    textlayout  :  textlayout
    device      :  bmp canon cmap cmapx cmapx_np dot emf emfplus eps fig gd gd2 gif gv imap imap_np ismap jpe jpeg jpg metafile pdf pic plain plain-ext png pov ps ps2 svg svgz tif tiff tk vml vmlz vrml wbmp xdot xdot1.2 xdot1.4
    loadimage   :  (lib) bmp eps gd gd2 gif jpe jpeg jpg png ps svg
```
But I'm still seeing the same output.

Is perhaps the issue I'm running into here `DOT_TRANSPARENT` is perhaps not being correctly read [here](https://github.com/doxygen/doxygen/blob/e18f715eb55121a4219d00bc4d824cebf1fb504b/src/dotgroupcollaboration.cpp#L332-L335)? Using `DOT_CLEANUP = NO` only seems to work if I introduce something that prevents `dot` from running successfully (I have just been sticking a `"` in the font name,) but when I do that I don't see the `bgcolor="transparent` line in any of the `.dot` files generated.

I don't have a development environment setup for doxygen, but if I can do that I'll run with a debugger attached and see if that line isn't executing as I think it should be.

**Comment by nitz** [comment]
I was!

But in an attempt to do it all again (while using `-d extcmd` to make sure), I have perhaps stumbled upon where my actual issue may lie! I've been using `doxywizard` as I've been tinkering through different settings as I'm setting up my project. It seems running doxygen via the `Run` tab isn't the same as running `doxygen` from the command line:

Comparing the two outputs:
![image](https://user-images.githubusercontent.com/602691/154518763-d603e287-1da5-44ab-aed7-a01660bc0bda.png)

Pressing `Show configuration` in doxywizard correctly calls out that `DOT_TRANSPARENT = YES`, but perhaps there's something along the way causing that setting to not make it through to the actual `doxygen` invocation?

**Comment by albert-github** [comment]
Which version of the doxywizard are you using (see the about box)?

**Comment by nitz** [comment]
1.9.3 w/Qt 5.15.2

![image](https://user-images.githubusercontent.com/602691/154529477-728e79ff-499d-4119-96db-91ae3b32f62c.png)

**Comment by albert-github** [comment]
Hmm, strange.

Very nice trick with `cat`!!

I remember we had problems in the past with something similar (see #8346, #8372) and that was solved. 
Looks like something similar, and with the `cat` it is easy/ier to reproduce.
I have to dive into it.

**Comment by albert-github** [comment]
I did quite a few experiments, but no real solution.
In the documentation of `QProcess` I see:

> Certain flags, such as Unbuffered and Truncate, are meaningless when used with some subclasses. Some of these restrictions are implied by the type of device that is represented by a subclass. In other cases, the restriction may be due to the implementation, or may be imposed by the underlying platform; for example, [QTcpSocket](https://doc.qt.io/qt-5/qtcpsocket.html) does not support Unbuffered mode, and limitations in the native API prevent [QFile](https://doc.qt.io/qt-5/qfile.html) from supporting Unbuffered on Windows.

Especially the last part ....
A solution like:
```
m_runProcess->start(doxygenPath + QString::fromLatin1("doxygen"), args, QIODevice::ReadWrite | QIODevice::Unbuffered);
```
instead of 
```
m_runProcess->start(doxygenPath + QString::fromLatin1("doxygen"), args);
```
which is equivalent to:
```
m_runProcess->start(doxygenPath + QString::fromLatin1("doxygen"), args, QIODevice::ReadWrite);
```
doesn't work.

The only solution I can, at the moment, think of is to change:
```
m_expert->writeConfig(t,false,false);
```
into
```
m_expert->writeConfig(t,true,true);
```
though this is just moving the problem a bit as for big project we still night run into problems.

Another possibility might be to invoke doxygen heer not with `-` (stdout), but with the saved doxygen configuration setting.

@doxygen What is your opinion?

**Comment by albert-github** [comment]
A correct solution looks like to be a bit easier than expected...

I've just pushed a proposed patch, pull request #9132

**Comment by nitz** [comment]
Wow. Of course it would be so simple. I didn't even think to look for a flush on the text stream 😅

Well done.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.4.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
